src/gpuarray_array.c

```diff
@@ -1117,8 +1121,7 @@ int GpuArray_is_c_contiguous(const GpuArray *a) {
   int i;
 
   for (i = a->nd - 1; i >= 0; i--) {
-    if (a->dimensions[i] != 1 && a->strides[i] != size)
-      return 0;
+    if (a->strides[i] != size) return 0;
```

Comment:
I find that too restrictive. What about:
if (a->strides[i] != size) continue;
else if(a->dimensions[i] == 1 && a->strides[i] % size == 0) continue;
else return 0;